src/tasks.ts

```diff
@@ -125,12 +125,18 @@ function getDataWithTimestampsAndGeoPoints(
       };
     }
     const value = Array.isArray(currData)
-      ? currData.map((dataItem) =>
-          convertValueToTimestampOrGeoPointIfPossible(
-            dataItem,
-            firestoreStatics,
-          ),
-        )
+      ? currData
+          .map((dataItem) =>
+            convertValueToTimestampOrGeoPointIfPossible(
+              dataItem,
+              firestoreStatics,
+            ),
+          )
+          .map((row) =>
+            row.constructor === Object
```

Comment:
Nice work handling the nested case, but I'm wondering - could this be part of a single map so that we aren't looping over the list twice?

Reply:
Can be done.
However, the abstractions in the first map function are not aligned.
In other words, it becomes difficult to understand.

I fixed.